cibyl/orchestrator.py

```diff
@@ -192,6 +158,53 @@ def setup_sources(self):
                         if source.enabled:
                             source.setup()
 
+    def load_features(self):
+        """Read user-requested features and setup the right argument to query
+        the information for them."""
+        features_loader = FeaturesLoader()
+        user_features = self.parser.ci_args.get('features')
+        if not user_features:
+            return []
+        if user_features and not user_features.value:
+            # throw error in case cibyl is called with --features argument but
+            # without any specified feature
+            msg = "No feature specified. Choose one of the following "
+            msg += "features:"
+            msg += features_loader.get_string_all_features()
+            raise InvalidArgument(msg)
+        return [features_loader.get_feature(feature_name)
+                for feature_name in user_features.value]
+
+    def run_features(self, system, features_to_run):
+        """Run user-requested features."""
+        features_combination = None
+        for feature_to_run in features_to_run:
+            feature_info = feature_to_run.query(system,
+                                                **self.parser.ci_args,
+                                                **self.parser.app_args)
+            if feature_info is None:
+                # no successful query was performed
+                system.add_feature(Feature(feature_to_run.name,
+                                           "N/A"))
+                # set the returned info to an empty container to have an empty
+                # intersection
+                feature_info = AttributeDictValue("name")
+            else:
+                system.add_feature(Feature(feature_to_run.name,
+                                           bool(feature_info)))
+
+            if "jobs" in self.parser.ci_args:
+                if features_combination is None:
+                    features_combination = feature_info
+                else:
+                    features_combination = intersect_models(
+                            features_combination, feature_info)
+
+        if features_combination:
```

Comment:
I believe this 'if' will never not happen as long as 'feature_info' always intersects.

Reply:
You're right, here the check should be if 'jobs' is in the ci_args, because otherwise it would be None. I've changed the condition to make it more explicit.